managed identities for quarkus and small doc changes (#32)
Azure-Samples/azure-openai-rag-workshop: docs/sections/java-quarkus/06-chat-api.md

As those producers are configured in separate files, and use the LangChain4J API, they can later be switched easily to use other implementations: this will be useful for example to use a more powerful language or embedding model, or for running tests locally.

Let's start by configuring `ChatLanguageModelAzureOpenAiProducer`, using the Azure OpenAI API.

#### Managing Azure credentials

> Before we can create the clients, we need to retrieve the credentials to access our Azure services. We'll use the [Azure Identity SDK](https://learn.microsoft.com/en-us/java/api/com.azure.identity?view=azure-java-stable) to do that.

Reviewer: ```suggestion
Before we can create the clients, we need to retrieve the credentials to access our Azure services. We'll use the [Azure Identity SDK](https://learn.microsoft.com/java/api/com.azure.identity?view=azure-java-stable) to do that.
```

Make sure this import is at the top of the file:

```java
import com.azure.identity.DefaultAzureCredentialBuilder;
```

> Then add this code to retrieve the token to build the `AzureOpenAIChatModel`

Reviewer: ```suggestion
Then add this code to retrieve the token to build the `AzureOpenAIChatModel`:
```

```java
AzureOpenAiChatModel model;

try {
  // Use the current user identity to authenticate with Azure OpenAI.

>   // (no secrets needed, just use 'az login' locally, and managed identity when deployed on Azure).

Reviewer: ```suggestion
  // (no secrets needed, just use `az login` or `azd auth login` locally, and managed identity when deployed on Azure).
```
